urllib3/connection.py

```diff
@@ -162,6 +162,37 @@ def connect(self):
         conn = self._new_conn()
         self._prepare_conn(conn)
 
+    def request_chunked(self, method, url, body=None, headers={}):
+        """
+        Alternative to the common request method, which sends the
+        body with chunked encoding and not as one block
+        """
+        header_names = set(k.lower() for k in headers)
```

Comment:
HTTPHeaderDict also has to lower everything internally, plus adds a lot of other overhead. I'd be a big fan of reusing HTTPHeaderDict throughout requests+urllib3, but as long as this doesn't happen consistently, there is nothing gained from having it here.